### PR TITLE
Feat(CI): Use  environment variable to specify the location of testing data

### PR DESCRIPTION
**Pull request by acezen**
### Reason for this PR

as issue #510 describes, since the released source code tarball would not contains the submodules, to make the test work with tarball, it's better to deprecate testing data submodule and use developer setting environment variable to indicate the location of testing data.

### What changes are included in this PR?

- refine the unit test of c++/scala library with the 'GAR_TEST_DATA' env variable
- update the related developer document 
- [TODO] Drop the testing submodule until java/python library update the unit test like cpp/scala. (java/python not bump up in this release)

### Are these changes tested?

already tested by current test case.

### Are there any user-facing changes?

no

### CODE REVIEW

**Comment by acezen** [comment]
Hi， @lixueclaire @SemyonSinchenko ， This PR may need your review, thanks.

**Review by lixueclaire — APPROVED** [review]
LGTM

File: cpp/README.md [hunk 1]
```diff
@@ -88,21 +88,27 @@ Debug build with unit tests:
...
     $ cmake -DCMAKE_BUILD_TYPE=Debug -DBUILD_TESTS=ON ..
     $ make -j8       # if you have 8 CPU cores, otherwise adjust, use -j`nproc` for all cores
-    $ make test      # to run the tests
+```
+
+After building, you can run the unit tests with:
+
+```bash
+    $ git clone https://github.com/apache/incubator-graphar-testing.git testing  # download the testing data
+    $ GAR_TEST_DATA=${PWD}/testing ctest
```
Comment (SemyonSinchenko) [inline cpp/README.md:97]: I would recommend to create a folder `dev/` and provide bash/python scripts for downloading the data and set up all the variables. We can do it in a separate PR. It is done like this in [spark](https://github.com/apache/spark/tree/master/dev) and I found it very user-friendly.

**Review by SemyonSinchenko — APPROVED** [review]

File: cpp/README.md (hunk 1 above)
Comment (acezen) [reply cpp/README.md:97]: The suggestion is great, I can do it in the PR #507 .
